test/unit/sourceset.test.js

```diff
@@ -276,14 +300,260 @@ QUnit[qunitFn]('sourceset', function(hooks) {
         validateSource(assert, this.player, [this.sourceOne, this.sourceTwo]);
         done();
       });
+
+      this.player.src([this.sourceOne, this.sourceTwo]);
     });
 
-    QUnit.test('no source', function(assert) {
+    QUnit.test('mediaEl.src = ...;', function(assert) {
       const done = assert.async();
 
-      this.player = videojs(this.mediaEl, {
-        enableSourceset: true
+      this.player = videojs(this.mediaEl, {enableSourceset: true});
+
+      this.player.one('sourceset', (e) => {
+        validateSource(assert, this.player, [this.testSrc]);
+        done();
+      });
+
+      this.player.tech_.el_.src = this.testSrc.src;
+    });
+
+    QUnit.test('mediaEl.setAttribute("src", ...)"', function(assert) {
+      const done = assert.async();
+
+      this.player = videojs(this.mediaEl, {enableSourceset: true});
+
+      this.player.one('sourceset', (e) => {
+        validateSource(assert, this.player, [this.testSrc]);
+        done();
+      });
+
+      this.player.tech_.el_.setAttribute('src', this.testSrc.src);
+    });
+
+    const appendTypes = [
+      {name: 'appendChild', fn: (el, obj) => el.appendChild(obj)},
+      {name: 'innerHTML', fn: (el, obj) => {el.innerHTML = obj.outerHTML;}}, // eslint-disable-line
+    ];
+
+    // ie does not support this and safari < 10 does not either
+    if (window.Element.prototype.append) {
+      appendTypes.push({name: 'append', fn: (el, obj) => el.append(obj)});
+    }
+
+    if (window.Element.prototype.insertAdjacentHTML) {
+      appendTypes.push({name: 'insertAdjacentHTML', fn: (el, obj) => el.insertAdjacentHTML('afterbegin', obj.outerHTML)});
+    }
+
+    appendTypes.forEach((appendObj) => {
+
+      QUnit.test(`<source> one source through ${appendObj.name}`, function(assert) {
+        const done = assert.async();
+
+        this.source = document.createElement('source');
+        this.source.src = this.testSrc.src;
+        this.source.type = this.testSrc.type;
+
+        this.player = videojs(this.mediaEl, {enableSourceset: true});
+
+        this.player.one('sourceset', (e) => {
+          validateSource(assert, this.player, [this.testSrc], false);
```

Comment:
I'm concerned that these tests that don't assert inside of `validateSource` (the last false being passed in), don't actually verify things we want verified, at least if that's the only call to `validateSource` without another assertion.